src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -466,6 +466,26 @@ internal static SecurityKey ResolveTokenSigningKey(string kid, string x5t, Token
             else
                 return null;
         }
+
+        /// <summary>
+        /// Returns number of Jwt parts in token.
+        /// </summary>
+        /// <param name="token">Input token to count patrs.</param>
+        /// <returns>Value between 1 and <see cref="JwtConstants.MaxJwtSegmentCount"/> + 1.</returns>
+        internal static int CountTokenJwtParts(string token)
```

Comment:
@lechu445 we need to be thinking about what to do if the count is greater than 5.
We can't just return as the jwt is not well formed.

Reply:
I've just added method DetectTokenType that returns Unknown enum value in case of not well formed token. The consumers of this method can handle the Unknown type on their way.